automation_tools/__init__.py

```diff
@@ -1348,13 +1405,23 @@ def product_install(distribution, create_vm=False, certificate_url=None,
                 execute(setup_oscap, host=host)
             if satellite_version not in ('6.0', '6.1'):
                 execute(oscap_content, host=host)
-            # if we have PXE default template we can setup foreman discovery
-            if os.environ.get('PXE_DEFAULT_TEMPLATE_URL') is not None:
-                execute(setup_foreman_discovery, host=host)
+            if satellite_version in ('6.1', '6.2'):
+                # if we have PXE default template we can setup discovery
+                if os.environ.get('PXE_DEFAULT_TEMPLATE_URL') is not None:
+                    # setup_foreman_discovery is applicable for 6.1 and 6.2
+                    execute(setup_foreman_discovery, host=host)
         # ostree plugin is for Sat6.2+ and nightly (rhel7 only)
         if satellite_version not in ('6.0', '6.1'):
             execute(
                 enable_ostree, distribution.endswith('upstream'), host=host)
+        # setup_foreman_discovery_70 is for upstream and upcoming 6.3 release
+        if satellite_version not in ('6.0', '6.1', '6.2'):
```

Comment:
Yeah.. completely forgot about `6.3`. Since we need this on priority so will make changes as per your suggestion.